lib/iris/coord_systems.py

```diff
@@ -791,6 +791,15 @@ def __repr__(self):
                    self.false_easting, self.false_northing,
                    self.secant_latitudes, self.ellipsoid)
 
+    def __eq__(self, other):
+        sattrs = self.__dict__.copy()
+        oattrs = other.__dict__.copy()
+        s_secant_latitudes = sattrs.pop('secant_latitudes', None)
+        o_secant_latitudes = oattrs.pop('secant_latitudes', None)
+        return (self.__class__ == other.__class__ and
+                sattrs == oattrs and
+                (s_secant_latitudes==o_secant_latitudes).all())
```

Comment:
I think you are right that the best solution is ensuring a tuple for the `secant_latitudes`. I will implement that and remove the comparison from the `LambertConformal` class.